### DISCUSSION TITLE
Environment variables in playwright tests

### DISCUSSION THREAD

**Discussion by jsheffers** [cawa-93/vite-electron-builder#514] [Q&A] [answered]
Is there a way to pipe in the VITE_ variables to the Playwright tests?  For example, depending on which Vite `MODE` is set, I load a different set of environment variables including the URL.  If I `console.log(process.env)` in `app.spec.js` none of the Vite variables are included.  I'm using this as a yarn command to start the test `MODE=mymode node tests/app.spec.js`.  Any thoughts?

**Comment by SalahAdDin** [comment]
Did you get any way to do this?

**Reply by cawa-93** [reply]
Try my [answer below](https://github.com/cawa-93/vite-electron-builder/discussions/514#discussioncomment-4708434)

**Comment by cawa-93** [comment, marked as answer]
Use [`loadEnv`](https://vitejs.dev/guide/api-javascript.html#loadenv) helper from Vite to load Environment variables from `.env` file wherever you want. It read `.env`  file and setup variables in `process.env`

**Reply by SalahAdDin** [reply]
Do you have any example of this with full setup?

**Reply by cawa-93** [reply]
> Do you have any example of this with full setup?

No. Just call loadEnv from test entry file.

**Reply by SalahAdDin** [reply]
I just need to pass `import.meta.env.BASE_URL;` to `playwright`, but how can we set the same base `url` for both `vite` and `BASE_URL`?

**Reply by SalahAdDin** [reply]
Ok, I followed this [example](https://stackblitz.com/edit/vite-base-url-sfqcrv?file=vite.config.ts) and I could load all variables from `.env` files, so we have the same values for **playwright** and **vite**.

Thanks!

**Reply by codespearhead** [reply]
Here's how I set it up based on the answers above. No additional dependencies are needed.

My env file is located higher up in the directory tree, hence the "ENV_DIR" variable.

The variables are accessible in Playwright test files via "process.env".

```bash
# vite.config.ts

import { defineConfig } from "vite";

export const ENV_DIR = "./../../../";
export default defineConfig({ envDir: ENV_DIR })
```

```bash
# playwright.config.ts

import { defineConfig } from '@playwright/test';
import { loadEnv } from 'vite';
import { ENV_DIR } from "./vite.config";

export const ENV_DIR = "./../../../";
Object.assign(process.env, loadEnv('', ENV_DIR));

export default defineConfig({ fullyParallel: true });
```

**Comment by hatton** [comment]
Is it possible to just inject one variable, at runtime, without using `.env`?  Before vite, I just had
```
electron.launch({
      env: {
        E2E: "true",
      }
    });
```

Whereas with the vite approach, it sounds like have I have to have a special `.env` for e2e, and make a special build for e2e, in order to inject this one value. Hopefully I'm just confused.

**Reply by codespearhead** [reply]
You can do it via the "loadEnv" method from "vite" [1], but you don't have to. See this [2] answer.

[1] import { loadEnv } from 'vite';
[2] https://github.com/cawa-93/vite-electron-builder/discussions/514#discussioncomment-12619929
